lib/timers.js

```diff
@@ -521,6 +537,19 @@ function unrefdHandle(timer, now) {
   return true;
 }
 
+Timeout.prototype[Symbol.toPrimitive] = function() {
+  const id = this[async_id_symbol];
+  if (id in KNOWN_TIMERS !== true) {
+    KNOWN_TIMERS[id] = this;
+    const $close = this.close;
+    this.close = function () {
+      delete KNOWN_TIMERS[this[async_id_symbol]];
+      this.close = $close;
+      this.close();
+    }
```

Comment:
This won't pass linting due to missing semicolon. Also, can the function be factored out?

Reply:
it cannot be factored out since it closes over `$close`.

Reply:
`$close` is effectively always equal to `Timeout.prototype.close`, right?

Reply:
since it is mutable, that is not certain.

Reply:
@bmeck I think it’s fine to just ignore that. You could still run into it anyway, when `.close` is changed after the `[@@toPrimitive]()` call…

Reply:
if they replace it they are probably wrapping it, I'm already ignoring all the ways to avoid getter/setters but don't feel comfortable with completely removing `.close` this seems like one of the metrics people could be profiling handle lifetimes.